### PR TITLE
[snapshoty] refactor: improve user experience

### PR DESCRIPTION
**Pull request by amannocci**
## What is the change being made?

* Upgrade to node 20 instead of node 16.
* Upgrade all dependencies.
* Make secrets only relevant values. 

## Why is the change being made?

* Currently, `snapshoty` process is hard to debug due to secrets everywhere.

### CODE REVIEW

File: .github/actions/snapshoty/index.js [hunk 1]
```diff
@@ -24,18 +24,28 @@ async function run() {
...
+          core.setSecret(value);
+        }
       }
+
       // Special case so we can inject env variables
       if (key.startsWith("SNAPSHOTY_")) {
         let value = process.env[key];
         let variable = key.replace("SNAPSHOTY_", "")
         args.push('-e', `${variable}=${value}`);
-        core.setSecret(value);
```
Comment (amannocci) [inline .github/actions/snapshoty/index.js:38]: Note: We only use this feature in one place to forward a date at `elastic`.

File: .github/actions/snapshoty/index.js [hunk 2]
```diff
@@ -24,18 +24,28 @@ async function run() {
...
+    }
+
+    // Forward env vars
     Object.keys(process.env).forEach(function (key) {
       if (key.startsWith("GITHUB_") || key.startsWith("RUNNER_")) {
         let value = process.env[key];
         args.push('-e', `${key}=${value}`);
-        core.setSecret(value);
+
+        if (key === 'GITHUB_TOKEN') {
```
Comment (amannocci) [inline .github/actions/snapshoty/index.js:39]: Note: Only the github token should be considered as a secret.
It should already be a secret.

File: .github/actions/snapshoty/index.js [hunk 3]
```diff
@@ -24,18 +24,28 @@ async function run() {
       '-e', `GCS_PRIVATE_KEY_ID=${gcsPrivateKeyId}`,
       '-e', `GCS_PROJECT=${gcsProject}`
     ]
+
+    // GCS env vars are secrets
+    for (secret of [gcsClientEmail, gcsPrivateKey, gcsPrivateKeyId, gcsProject]) {
```
Comment (amannocci) [inline .github/actions/snapshoty/index.js:29]: Note: Let's consider those inputs as secrets.

File: .github/actions/snapshoty/action.yml [hunk 4]
```diff
@@ -20,5 +20,5 @@ inputs:
     description: 'Google Cloud project id of the service account'
     required: true
 runs:
-  using: 'node16'
+  using: 'node20'
```
Comment (amannocci) [inline .github/actions/snapshoty/action.yml:23]: Note: node 16 is considered EOL for a while.

**Review by v1v — APPROVED** [review]
LGTM

**Review by reakaleek — APPROVED** [review]

**Pull request merged**